src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.EasyRequest.cs

```diff
@@ -370,6 +370,10 @@ internal void SetPossibleRedirectForLocationHeader(string location)
                     {
                         SetCookieOption(newUri);
                     }
+
+                    if(newUri.Scheme == Uri.UriSchemeHttp && _requestMessage.RequestUri.Scheme == Uri.UriSchemeHttps) {
```

Comment:
Interestingly we could use this same check to avoid doing some of the above work if we're not going to redirect anyway.  But it's probably not something worth optimizing for.